src/Microsoft.ML.CodeGen/CodeGenerator/CSharp/CodeGeneratorSettings.cs

```diff
@@ -0,0 +1,39 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using Microsoft.ML.AutoML;
+
+namespace Microsoft.ML.CodeGenerator.CSharp
+{
+    public class CodeGeneratorSettings
+    {
+        public CodeGeneratorSettings()
+        {
+            // set default value
+            Target = GenerateTarget.Cli;
+        }
+
+        public string LabelName { get; set; }
+
+        public string ModelPath { get; set; }
+
+        public string OutputName { get; set; }
+
+        public string OutputBaseDir { get; set; }
+
+        public string TrainDataset { get; set; }
+
+        public string TestDataset { get; set; }
+
+        public enum GenerateTarget
```

Comment:
Hmm I'm surprised StyleCop would let you have enum intertwined with class members -- is this no longer a style guideline to have them above members?  It makes sense to have it close to the member that uses it but I think most people expect enums to be listed in their own section.

Reply:
move `GenerateTarget` out of `CodeGeneratorSettings` Class